File: ios/Plugin/SharetoPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SharetoPlugin)
public class SharetoPlugin: CAPPlugin {
    private let implementation = Shareto()

    @objc func shareImage(_ call: CAPPluginCall) {
        let base64 = call.getString("base64")
        let fileName = call.getString("fileName")
        let phoneNumber = call.getString("phoneNumber")
        let message = call.getString("message")
        guard let bundleId = call.getString("bundleId") else {
            call.reject("bundleId is required")
            return
        }
        
        guard let viewController = self.bridge?.viewController else {
            call.reject("Unable to get view controller")
            return
        }
        
        implementation.shareImage(
            base64: base64,
            fileName: fileName,
            phoneNumber: phoneNumber,
            message: message,
            bundleId: bundleId,
            viewController: viewController
        ) { error in
            if let error = error {
                call.reject(error.localizedDescription)
            } else {
                call.resolve()
            }
        }
    }
    
    @objc func sharePdf(_ call: CAPPluginCall) {
        let base64 = call.getString("base64")
        let fileName = call.getString("fileName")
        let phoneNumber = call.getString("phoneNumber")
        let message = call.getString("message")
        guard let bundleId = call.getString("bundleId") else {
            call.reject("bundleId is required")
            return
        }
        
        guard let viewController = self.bridge?.viewController else {
            call.reject("Unable to get view controller")
            return
        }
        
        implementation.sharePdf(
            base64: base64,
            fileName: fileName,
            phoneNumber: phoneNumber,
            message: message,
            bundleId: bundleId,
            viewController: viewController
        ) { error in
            if let error = error {
                call.reject(error.localizedDescription)
            } else {
                call.resolve()
            }
        }
    }
}

File: android/src/main/java/com/pareshgami/shareto/SharetoPlugin.java
package com.pareshgami.shareto;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import android.util.Log;

@CapacitorPlugin(name = "Shareto")
public class SharetoPlugin extends Plugin {

    private Shareto implementation;

    @Override
    public void load() {
        implementation = new Shareto(getContext());
    }

    @PluginMethod
    public void shareImage(PluginCall call) {
        String base64 = call.getString("base64");
        String fileName = call.getString("fileName");
        String phoneNumber = call.getString("phoneNumber");
        String message = call.getString("message");
        String bundleId = call.getString("bundleId");

        try {
            implementation.shareImage(base64, fileName, phoneNumber, message, bundleId);
            call.resolve();
        } catch (IllegalArgumentException e) {
            Log.e("SharetoPlugin", "Invalid argument: " + e.getMessage(), e);
            call.reject("Invalid argument: " + e.getMessage());
        } catch (Exception e) {
            Log.e("SharetoPlugin", "Unexpected error: " + e.getMessage(), e);
            call.reject("Unexpected error: " + e.getMessage());
        }
    }

    @PluginMethod
    public void sharePdf(PluginCall call) {
        String base64 = call.getString("base64");
        String fileName = call.getString("fileName");
        String phoneNumber = call.getString("phoneNumber");
        String message = call.getString("message");
        String bundleId = call.getString("bundleId");

        try {
            implementation.sharePdf(base64, fileName, phoneNumber, message, bundleId);
            call.resolve();
        } catch (IllegalArgumentException e) {
            Log.e("SharetoPlugin", "Invalid argument: " + e.getMessage(), e);
            call.reject("Invalid argument: " + e.getMessage());
        } catch (Exception e) {
            Log.e("SharetoPlugin", "Unexpected error: " + e.getMessage(), e);
            call.reject("Unexpected error: " + e.getMessage());
        }
    }
}
